Regulation API documentation updates (#555)
rudderlabs/rudderstack-docs: rudderstack-api/data-regulation-api.mdx

This guide covers the data regulation feature in detail and details the Data Regulation API endpoints.

<div class="warningBlock">

> The Data Regulation API is applicable only for the destinations that support events sent via the <a href="https://www.rudderstack.com/docs/rudderstack-cloud/rudderstack-connection-modes/#cloud-mode">RudderStack cloud mode</a>.

Reviewer: ```suggestion
The Data Regulation API is applicable only for the destinations configured to sent events using the <a href="https://www.rudderstack.com/docs/rudderstack-cloud/rudderstack-connection-modes/#cloud-mode">RudderStack cloud mode</a>.
```

Reviewer: I find `support` a bit confusing since amplitude supports both cloud and device mode, the issue is when device mode is being configured/used that we can not support it.
